R/module-repository.R

```diff
@@ -364,39 +382,48 @@ setMethod(
 setMethod(
   "downloadModule",
   signature = c(name = "character", path = "missing", version = "missing",
-                repo = "missing", data = "missing", quiet = "missing", quickCheck = "ANY"),
+                repo = "missing", data = "missing", quiet = "missing", quickCheck = "ANY",
+                overwrite = "missing"),
   definition = function(name, quickCheck) {
     files <- downloadModule(name, path = getOption("spades.modulePath"),
                             version = NA_character_,
                             repo = getOption("spades.moduleRepo"),
                             data = FALSE, quiet = FALSE,
-                            quickCheck = quickCheck)
+                            quickCheck = quickCheck, overwrite = FALSE)
     return(invisible(files))
 })
 
 #' @rdname downloadModule
 setMethod(
   "downloadModule",
   signature = c(name = "character", path = "ANY", version = "ANY",
-                repo = "ANY", data = "ANY", quiet = "ANY", quickCheck = "ANY"),
-  definition = function(name, path, version, repo, data, quiet, quickCheck) {
+                repo = "ANY", data = "ANY", quiet = "ANY", quickCheck = "ANY",
+                overwrite = "ANY"),
+  definition = function(name, path, version, repo, data, quiet, quickCheck,
+                        overwrite) {
     if (missing(path)) path <- getOption("spades.modulePath")
     if (missing(version)) version <- NA_character_
     if (missing(repo)) repo <- getOption("spades.moduleRepo")
     if (missing(data)) data <- FALSE
     if (missing(quiet)) quiet <- FALSE
+    if (missing(overwrite)) overwrite <- FALSE
 
-    files <- downloadModule(name, path, version, repo, data, quiet, quickCheck)
+    files <- downloadModule(name, path, version, repo, data, quiet, quickCheck, overwrite)
     return(invisible(files))
 })
 
 ################################################################################
 #' Download module data
 #'
 #' Download external data for a module if not already present in the module
-#' directory or if there is a checksum mismatch indicating that the file is not
+#' directory, or if there is a checksum mismatch indicating that the file is not
 #' the correct one.
 #'
+#' \code{downloadData} requires a checksums file to work, as it will only download
+#'  the files specified therein. Hence, module developers should make sure they
+#'  have manually downloaded all the necessary data and ran \code{checksums} to
+#'  build a checksums file.
+#'
```

Comment:
It's useful info (and perhaps it's mentioned in the `checksums` docs or in a vignette) but I'm not sure that this belongs here. Open to discussion.

Reply:
I don't think it's mentioned in the checkSums and the lack of info made me lose a few hours, because I just didn't understand why the heck downloadData wasn't working... So, yes open to discussion, BUT I still think that the fact that downloadData depends on the presence of a checksums file is not explicit in the present form of the doc.

Reply:
downloadData relies on sourceUrls specified in the module metadata. The checksums file is there to assist -- files can be checksummed that aren't downloaded directly. Also, the checksums file does'nt keep any info regarding the source of the files.

I agree that we need to make all of this more clear. We just need to decide where all this info should go.

Reply:
I'm confused. When I was trying to use `downloadData` in a module that I created it didn't work when the checksums file was missing, or was empty. @eliotmcintire then told me that I needed to manually put the data on the folder and create a checksums.txt so that `downloadData` could later check it and effectively get the data if need be.